src/models/disassemblymodel.cpp

```diff
@@ -0,0 +1,127 @@
+/*
+  disassemblymodel.cpp
+
+  This file is part of Hotspot, the Qt GUI for performance analysis.
+
+  Copyright (C) 2020 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+  Author: Darya Knysh <d.knysh@nips.ru>
+
+  Licensees holding valid commercial KDAB Hotspot licenses may use this file in
+  accordance with Hotspot Commercial License Agreement provided with the Software.
+
+  Contact info@kdab.com if any conditions of this licensing are not clear to you.
+
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "disassemblyoutput.h"
+
+DisassemblyModel::DisassemblyModel(QObject *parent)
+        : QAbstractTableModel(parent)
+{
+}
+
+void DisassemblyModel::setDisassembly(QVector<DisassemblyOutput::DisassemblyLine> data)
+{
+    m_data = data;
+}
+
+void DisassemblyModel::setResults(Data::CallerCalleeResults results)
+{
+    m_results = results;
+    m_numTypes = results.selfCosts.numTypes();
```

Comment:
remove and inline the call instead where appropriate

Reply:
Does the comment concern the whole function setResults? It is used in ResultsDisassemblyPage. I am afraid I didn't catch your thought.

Reply:
no, I'd remove the `m_numTypes` but then again, you use it in many places, so maybe keep it after all